### DISCUSSION TITLE
[Question]　'Articulation' object has no attribute 'actuators'

### DISCUSSION THREAD

**Discussion by kokoroA** [isaac-sim/IsaacLab#1135] [Q&A]
I am trying to modify the quadcopter Direct RL sample based on the [official documentation](https://isaac-sim.github.io/IsaacLab/source/tutorials/03_envs/modify_direct_rl_env.html) to train an original drone USD model. However, I am encountering the following error.

```python
File "/home/itolab/documents/IsaacLab_new/IsaacLab/source/extensions/omni.isaac.lab/omni/isaac/lab/assets/articulation/articulation.py", line 167, in reset
    for actuator in self.actuators.values():
AttributeError: 'Articulation' object has no attribute 'actuators'
```

Here is my `ArticulationCfg` and USD model hierarchy .

```python
robot = ArticulationCfg(
        prim_path="/World/envs/env_.*/Robot",
        spawn=sim_utils.UsdFileCfg(
            usd_path="/home/itolab/documents/Stampfly_parts/usd/Stampfly_joint.usd",
            rigid_props=sim_utils.RigidBodyPropertiesCfg(
            disable_gravity=False,
            max_depenetration_velocity=10.0,
            enable_gyroscopic_forces=True,
        ),
        articulation_props=sim_utils.ArticulationRootPropertiesCfg(
            enabled_self_collisions=False,
            solver_position_iteration_count=4,
            solver_velocity_iteration_count=0,
            sleep_threshold=0.005,
            stabilization_threshold=0.001,
        ),
        copy_from_source=False,
        ),
        init_state=ArticulationCfg.InitialStateCfg(
            pos=(0.0,0.0,0.5),
            joint_pos={
                ".*": 0.0,
            },
            joint_vel={
                "FL_joint":-200,
                "FR_joint":200,
                "RL_joint":200,
                "RR_joint":-200,
            },
        ),
        actuators={
            "dummy": ImplicitActuatorCfg(
                joint_names_expr=[".*"],
                stiffness=0.0,
                damping=0.0,
            ),
        }
    )
```

![Screenshot from 2024-10-04 15-52-57](https://github.com/user-attachments/assets/5aa54e30-a513-4b96-a93f-54e567f2d943)

And here is log file

[kit_20241004_144910.log](https://github.com/user-attachments/files/17254638/kit_20241004_144910.log)

Does anyone know the solutions about this?
Thank you in advance.

**Comment by Saumya-Karan** [comment]
def __init__(self, cfg: ArticulationCfg):
    """Initialize the articulation.

    Args:
        cfg: A configuration instance.
    """
    super().__init__(cfg)
    self.actuators = {}  # Initialize actuators as an empty dictionary
    self._has_implicit_actuators = False  # Initialize the attribute to prevent AttributeError


This solves the error in my case.

**Reply by Zhenghua-NCSU** [reply]
Thank you for your answer, I tried it the way you did but it doesn't work, can you help me to see if I modified the code correctly?


class HexaEnv(LocomotionEnv):
    cfg: HexaEnvCfg

    def __init__(self, cfg: HexaEnvCfg, render_mode: str | None = None, **kwargs):
        super().__init__(cfg, render_mode, **kwargs)
        self.actuators = {}  # Initialize actuators as an empty dictionary
        self._has_implicit_actuators = False  # Initialize the attribute to prevent AttributeError

The content of the error is exactly the same as the publisher's error.

**Reply by zxfever** [reply]
hey bro, face the same error, did you fix it?

**Comment by dantes-ai** [comment]
u have this in ur setup scene?

def _setup_scene(self):
    self.robot = self.cfg.robot.class_type(self.cfg.robot)
    self.scene.articulations["robot"] = self.robot

**Comment by zxfever** [comment]
hey bro, face the same error, did you fix it?

**Comment by Mayankm96** [comment]
Can you please share the kit log file?

> https://isaac-sim.github.io/IsaacLab/main/source/refs/troubleshooting.html#checking-the-internal-logs-from-the-simulator

**Comment by XZDang13** [comment]
The main issue is how you define the init_state.
The value of joint_vel should be float not int. 
joint_vel={
                "FL_joint":-200.0,
                "FR_joint":200.0,
                "RL_joint":200.0,
                "RR_joint":-200.0,
            },
        ),

For any value in init_state, I guess they should all be float.
